dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/SubsystemFeatureDefinitionImpl.java

```diff
@@ -484,6 +484,33 @@ public String toString() {
         return icons;
     }
 
+    /*
+     * (non-Javadoc)
+     *
+     * @see com.ibm.ws.kernel.feature.provisioning.ProvisioningFeatureDefinition#getEndpointIcons()
+     */
+    @Override
+    public Collection<String> getEndpointIcons() {
+        Collection<String> icons = new ArrayList<String>();
```

Comment:
Because this is rare, this ArrayList should only be constructed if endpointIconsHeader  is non-null.  So assign `icons` to Collections.emptyList here and then construct the ArrayList inside the if NOT null.

Reply:
We merged getEndpointIcons() behavior into getIcons(), which obviates this change.